src/main/java/net/neoforged/neoforge/event/AddReloadListenerEvent.java

```diff
@@ -5,49 +5,32 @@
 
 package net.neoforged.neoforge.event;
 
-import com.google.common.collect.ImmutableList;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.Executor;
-import net.minecraft.core.HolderLookup;
 import net.minecraft.core.RegistryAccess;
+import net.minecraft.resources.ResourceLocation;
 import net.minecraft.server.ReloadableServerResources;
 import net.minecraft.server.packs.resources.PreparableReloadListener;
-import net.minecraft.server.packs.resources.ResourceManager;
-import net.neoforged.bus.api.Event;
-import net.neoforged.fml.ModLoader;
 import net.neoforged.neoforge.common.NeoForge;
 import net.neoforged.neoforge.common.conditions.ICondition;
-import net.neoforged.neoforge.resource.ContextAwareReloadListener;
+import net.neoforged.neoforge.resource.VanillaServerListeners;
+import org.jetbrains.annotations.ApiStatus;
 
 /**
  * The main ResourceManager is recreated on each reload, just after {@link ReloadableServerResources}'s creation.
  *
  * The event is fired on each reload and lets modders add their own ReloadListeners, for server-side resources.
  * The event is fired on the {@link NeoForge#EVENT_BUS}
  */
-public class AddReloadListenerEvent extends Event {
-    private final List<PreparableReloadListener> listeners = new ArrayList<>();
+public class AddReloadListenerEvent extends SortedReloadListenerEvent {
```

Comment:
Should the name not contain `Server` somehow?

Reply:
It could, but I'm okay with the current name. I prefer to imply `Server` as a default state that doesn't need to be present in the class name, while client-only is non-default and must be reflected (hence `AddClientReloadListenerEvent`).

Reply:
A silly example: we have ServerStartedEvent not StartedEvent. The event is limited to server reload listeners and the naming should make it clear. Even the documentation mentions `server-side resources`.

Reply:
Updated to `AddClientReloadListenersEvent` / `AddServerReloadListenersEvent` (explicit sidedness + plural)